src/formatDate.js

```diff
@@ -8,7 +8,24 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = Array.from(fromFormat).pop();
+  const newFormat = Array.from(toFormat);
+  const newSeparator = newFormat.pop();
+  const dataObject = {};
+
+  for (const value of date.split(oldSeparator)) {
+    const key = fromFormat.shift();
+
+    dataObject[key] = value;
+
+    if (key === 'YY') {
+      dataObject.YYYY = value < 30 ? '20' + value : '19' + value;
+    } else if (key === 'YYYY') {
+      dataObject.YY = value.substring(2);
+    }
+  }
```

Comment:
The current implementation modifies the `fromFormat` array by using `shift()`, which will cause issues if `formatDate` is called multiple times with the same `fromFormat` array. It's better to use a separate index variable to access the corresponding format parts without modifying the original array.